### DISCUSSION TITLE
GLTF Scene Loading causes Freeze to entire Patch

### DISCUSSION THREAD

**Discussion by strangerattractor** [cables-gl/cables_docs#436] [! Welcome - readme first!]
Also made a test Patch for this: https://cables.gl/p/639f6fd41cbd5dac9fabd106?s=gR0vStZ0ghtMsOpI

When switching between certain GLTF-Scenes,  the entire Browser freezes for a Moment. 
Pandur mentioned that it might have to do with missing tangents that need to be calculated for the model. 

Curious in the testcase: Model 0 has 52770 Primitives, and Model 1 has 18918 Primitives..., but loading 
Model 1 causes the freeze. 

Also Model 4 has issues, when loading the texture. This also causes the browser to freeze for a moment. 

What can I do to avoid these freezes?

**Comment by pandrr** [comment]
one is compressed using draco, which is sadly quite cpu and memory intense to decompress...

that saying on a m1 macbook pro it does not freeze at all...

please try without draco and let me know if this is the problem

in the structure view you can now see what takes how long, but it will not show up there because loading it using draco is asynchronous and we have little control over it and can't reallty measure...

**Reply by pandrr** [reply]
if you want to reduce file size: make sure your server is using gzip for glb files
